src/theory/uf/equality_engine.cpp

```diff
@@ -926,7 +926,7 @@ std::string EqualityEngine::edgesToString(EqualityEdgeId edgeId) const {
   return out.str();
 }
 
-void EqualityEngine::explainEquality(TNode t1, TNode t2, bool polarity, std::vector<TNode>& equalities, EqProof * eqp) const {
+void EqualityEngine::explainEquality(TNode t1, TNode t2, bool polarity, std::vector<TNode>& equalities, std::shared_ptr<EqProof> eqp) const {
```

Comment:
It seems like `std::shared_ptr<EqProof> eqp` should be a `std::weak_ptr` since `explainEquality` is not taking (shared) ownership.